orchagent/switchorch.cpp

```diff
@@ -127,6 +151,55 @@ void SwitchOrch::doCfgSensorsTableTask(Consumer &consumer)
     }
 }
 
+
+bool SwitchOrch::switchTunnelSetVxlanParams(swss::FieldValueTuple i)
+{
+    auto attribute = fvField(i);
+    auto value = fvValue(i);
+    sai_attribute_t attr;
+    sai_status_t status;
+
+    if (!m_userSportModeEnabled)
+    {
+        // Enable Vxlan src port range feauture
+        attr.id = SAI_SWITCH_TUNNEL_ATTR_TUNNEL_VXLAN_UDP_SPORT_MODE;
+        attr.value.s32 = SAI_TUNNEL_VXLAN_UDP_SPORT_MODE_USER_DEFINED;
+        status = sai_switch_api->set_switch_tunnel_attribute(switch_tunnel_id, &attr);
+
+        m_userSportModeEnabled = true;
+
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            SWSS_LOG_ERROR("Failed to set SAI_TUNNEL_VXLAN_UDP_SPORT_MODE_USER_DEFINED  src port mode rv:%d",  status);
+            return handleSaiSetStatus(SAI_API_SWITCH, status) == task_need_retry;
+        }
+    }
+
+    attr.id = switch_tunnel_attribute_map.at(attribute);
+    switch (attr.id)
+    {
+        case SAI_SWITCH_TUNNEL_ATTR_VXLAN_UDP_SPORT:
+            attr.value.u16 = to_uint<uint16_t>(value);
+            break;
+        case SAI_SWITCH_TUNNEL_ATTR_VXLAN_UDP_SPORT_MASK:
+            attr.value.u8 = to_uint<uint8_t>(value);
+            break;
+        default:
+            SWSS_LOG_ERROR("Invalid switch tunnel attribute id %d", attr.id);
+            return false;
+    }
+
+    status  = sai_switch_api->set_switch_tunnel_attribute(switch_tunnel_id, &attr);
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Failed to set tunnnel switch attribute %s to %s, rv:%d", attribute.c_str(), value.c_str(), status);
+        return handleSaiSetStatus(SAI_API_SWITCH, status) == task_need_retry;
+    }
+
+    SWSS_LOG_NOTICE("Set switch attribute %s to %s", attribute.c_str(), value.c_str());
+    return false;
```

Comment:
looks like should be ```return true;```

Reply:
from my perspective we should return **true** when function fails and we need to retry. **false** - otherwise when everything is good